bindinfo/handle.go

```diff
@@ -400,35 +400,36 @@ func (tmpMap *tmpBindRecordMap) flushToStore() {
 	tmpMap.Store(newMap)
 }
 
-func (tmpMap *tmpBindRecordMap) saveToCache(bindRecord *BindRecord) {
+// Add puts a BindRecord into tmpBindRecordMap.
+func (tmpMap *tmpBindRecordMap) Add(bindRecord *BindRecord) {
 	key := bindRecord.OriginalSQL + ":" + bindRecord.Db + ":" + bindRecord.Bindings[0].ID
 	if _, ok := tmpMap.Load().(map[string]*bindRecordUpdate)[key]; ok {
 		return
 	}
 	tmpMap.Lock()
 	defer tmpMap.Unlock()
-	if _, ok := tmpMap.Load().(map[string]*bindRecordUpdate)[key]; ok {
+	newMap := copyBindRecordUpdateMap(tmpMap.Load().(map[string]*bindRecordUpdate))
+	if _, ok := newMap[key]; ok {
```

Comment:
It's better to check whether the new bind record exists. Which can save a memory copy overhead when it's existed.